Gifski/TrimmingAVPlayerView.swift

```diff
@@ -0,0 +1,47 @@
+import AVKit
+
+final class TrimmingAVPlayerView: AVPlayerView {
+	private var timeRangeObserver: NSKeyValueObservation?
+
+	func observeTrimmedTimeRange(_ updateClosure: @escaping (ClosedRange<Double>) -> Void) {
+		timeRangeObserver = player?.currentItem?.observe(\.duration, options: [.new]) { item, _ in
+			let startTime = item.reversePlaybackEndTime.seconds
+			let endTime = item.forwardPlaybackEndTime.seconds
+			if !startTime.isNaN && !endTime.isNaN {
```

Comment:
Can it ever be NaN or are you just being cautious?

Reply:
yeah, sometimes it returns noisy data so I had to do this otherwise it would crash